### PR TITLE
General PR slow CI

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Extend #30341 to handle general PR slow CI. Now, it could handle

- new model pull request
- or a pull request with the (latest) commit have the form `[run-slow]bert,gpt2`
  - the prefix could be `[run-slow]`, `[run_slow]` or `[run slow]`
  - the remaining part should be comma-separated like `bert,gpt2` or `bert, gpt2` etc.
    - there is no syntax check or error handling of the wrong format (we can add them later if we find necessary)

The script and workflow file names (and their methods or job/step names too) to reflect they handle more general cases instead of just new model cases.

[An example run](https://github.com/huggingface/transformers/actions/runs/8896080103/job/24427938756):
  - (I deliberate make a failing tests)
  - it's trigger with the commit message `[run-slow]bert,gpt2`

IMPORTANT: the label to add in order to trigger this CI is **run-slow**. 

`single-model-run-slow` is no longer used.

### CODE REVIEW

File: .github/workflows/self-pr-slow-ci.yml [hunk 1]
```diff
@@ -4,6 +4,11 @@ on:
   pull_request:
     paths:
       - "src/transformers/models/*/modeling_*.py"
+      - "tests/models/*/test_*.py"
```
Comment (ydshieh) [inline .github/workflows/self-pr-slow-ci.yml:7]: If nothing changed in those paths, the CI won't be triggered at all
(even if the commit message specifies some models)

File: .github/workflows/self-pr-slow-ci.yml [hunk 2]
```diff
@@ -4,6 +4,11 @@ on:
...
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
```
Comment (ydshieh) [inline .github/workflows/self-pr-slow-ci.yml:11]: To cancel the CI runs trigger by previous commits of the same PR

File: .github/workflows/self-pr-slow-ci.yml [hunk 3]
```diff
@@ -20,31 +25,46 @@ env:
...
+            echo "commit_message=$(git show -s --format=%s)" >> $GITHUB_ENV
 
-        - name: Check if there is a new model
-          id: check_new_model
+        - name: Get models to run slow tests
           run: |
+            echo "${{ env.commit_message }}"
             python -m pip install GitPython
-            echo "new_model=$(python utils/check_if_new_model_added.py | tail -n 1)" >> $GITHUB_OUTPUT
+            python utils/pr_slow_ci_models.py --commit_message "${{ env.commit_message }}" | tee output.txt
```
Comment (ydshieh) [inline .github/workflows/self-pr-slow-ci.yml:49]: now that (renamed) script will give the models to run (either new model or from the commit message)

**Review by amyeroberts — APPROVED** [review]
Amazing ❤️  Thanks for adding this so quickly - looking forward to trying it out!

File: utils/pr_slow_ci_models.py [hunk 4]
```diff
@@ -93,4 +98,48 @@ def get_new_python_files() -> List[str]:
...
+        command = command_search.groups()[0]
+        command = command.lower().replace("-", " ").replace("_", " ")
+        run_slow = command == "run slow"
+        if run_slow:
+            models = command_search.groups()[1].strip()
+            return models
+        else:
+            return ""
+    else:
+        return ""
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply utils/pr_slow_ci_models.py]: kinda copy paste from test fetcher, but yes your point is nice!

File: .github/workflows/self-pr-slow-ci.yml (hunk 1 above)
Comment (ydshieh) [reply .github/workflows/self-pr-slow-ci.yml:7]: Yes, we will see 🚀 !

**Pull request merged**